### PR TITLE
Upgrade and add missing rules

### PR DESCRIPTION
**Pull request by DaleLJefferson**
- comma-dangle
- no-return-await
- no-useless-return
- prefer-promise-reject-errors
- require-await
- prefer-destructuring

### CODE REVIEW

**Comment by DaleJefferson** [comment]
Just a test

**Comment by DaleJefferson** [comment]
First/nSecond

**Comment by DaleJefferson** [comment]
First
Second

**Comment by DaleJefferson** [comment]
- First
- Second

**Comment by DaleJefferson** [comment]
# Warnings: 
- First
- Second

**Comment by DaleJefferson** [comment]
- Warnings: 
 - First
 - Second

**Comment by DaleJefferson** [comment]
- Warnings: 
  - First
  - Second

**Comment by DaleJefferson** [comment]
## First ### 
 This is the first message/n/n## Second ### 
 This is the first message

**Comment by DaleJefferson** [comment]
## First ### 
 This is the first message
## Second ### 
 This is the first message

**Comment by DaleJefferson** [comment]
## First ### 
 This is the first message
## Second ### 
 This is the first message

 Created by PR-Check

**Comment by DaleJefferson** [comment]
## First ### 
 This is the first message
## Second ### 
 This is the first message

_Created_ by PR-Check_

**Comment by DaleJefferson** [comment]
## First ### 
 This is the first message
## Second ### 
 This is the first message

_Created by PR-Check_

**Comment by DaleJefferson** [comment]
## First ### 
 This is the first message
## Second ### 
 This is the second message
## Third ### 
 This is the third message

---
_Created by PR-Check_

File: eslint.js [hunk 1]
```diff
@@ -152,10 +154,13 @@ module.exports = {
         'no-useless-call': 'error',
         'no-useless-concat': 'error',
         'no-useless-escape': 'error',
+        'no-useless-return': 'error',
```
Comment (fjmorant) [inline eslint.js:157]: Good!! I find useless returns totally redudant

File: eslint.js [hunk 2]
```diff
@@ -152,10 +154,13 @@ module.exports = {
...
         'no-void': 'error',
         'no-warning-comments': 'error',
         'no-with': 'error',
+        'prefer-promise-reject-errors': 'error',
         radix: 'error',
+        'require-await': 'error',
```
Comment (fjmorant) [inline eslint.js:163]: 👍

File: eslint.js [hunk 3]
```diff
@@ -234,8 +239,12 @@ module.exports = {
         'max-depth': ['error', 2],
         'max-len': ['error', {
             code: 120,
+            ignoreComments: true,
             ignoreTrailingComments: true,
             ignoreUrls: true,
+            ignoreStrings: true,
+            ignoreTemplateLiterals: true,
```
Comment (fjmorant) [inline eslint.js:246]: Nice one

**Comment by jdmunro** [comment]
We ought to explain why we've added some of these rules. I'm sure they're valuable but the first thing devs are going to notice is potential errors whilst linting a project.

File: eslint.js [hunk 4]
```diff
@@ -325,6 +335,12 @@ module.exports = {
         'object-shorthand': ['error', 'always'],
         'prefer-arrow-callback': 'error',
         'prefer-const': 'error',
+        'prefer-destructuring': ['error', {
+            array: true,
```
Comment (fjmorant) [inline eslint.js:339]: Does this mean we can't destructure a function?

**Comment by DaleLJefferson** [comment]
I agree @jdmunro, the more I thought about it the more I saw the overlap with the handbook, I think we should add to the handbook and make sure everyone agrees before adding the ESLint rules.

I added these over the weekend, maybe someone could add reasons to the handbook

**Review by fjmorant — APPROVED** [review]
👍

**Pull request merged**